### PR TITLE
Improved DPI change handling

### PR DESCRIPTION
**Pull request by TellowKrinkle**
- Windows now get notified when they move between monitors with different DPIs
  - MoltenVK no longer gets squished into a tiny corner of the window or expanded to 2x the window size when moving
- DPI changes are now reflected in imgui, and imgui no longer explodes if you set its dpi more than once
  - Metal is currently the only renderer that fully notices DPI changes and tells imgui about it.  I'll deal with the others some other time.

### CODE REVIEW

File: Source/Core/VideoCommon/OnScreenUI.cpp [hunk 1]
```diff
@@ -343,6 +342,8 @@ void OnScreenUI::SetScale(float backbuffer_scale)
   ImGui::GetIO().DisplayFramebufferScale.x = backbuffer_scale;
   ImGui::GetIO().DisplayFramebufferScale.y = backbuffer_scale;
   ImGui::GetIO().FontGlobalScale = backbuffer_scale;
+  ImGui::GetStyle() = {};
```
Comment (MayImilae) [inline Source/Core/VideoCommon/OnScreenUI.cpp:345]: This doesn't look right. What are you trying to do here exactly?

Comment (TellowKrinkle) [reply Source/Core/VideoCommon/OnScreenUI.cpp:345]: ScaleAllSizes scales the existing sizes.  So if you call it twice with 2, you get 4x.  Call it three times, and you get 8x.

Should I add a comment, or would it make more sense to separate the change into a separate commit with a message describing the issue there?

Comment (iwubcode) [reply Source/Core/VideoCommon/OnScreenUI.cpp:345]: Definitely feels comment worthy to me

Comment (TellowKrinkle) [reply Source/Core/VideoCommon/OnScreenUI.cpp:347]: Done

**Review by iwubcode — APPROVED** [review]
Untested.  Code LGTM!

File: Source/Core/DolphinQt/RenderWidget.cpp
Comment (phire) [inline Source/Core/DolphinQt/RenderWidget.cpp:448]: Results in undefined behaviour:

```
/Users/phire/dolphin/dolphin_emu/Source/Core/DolphinQt/RenderWidget.cpp:477:30: 
runtime error: downcast of address 0x00016b44d9f0 which does not point to an object of type 'QResizeEvent'
0x00016b44d9f0: note: object is of type 'QPaintEvent'
 b0 04 60 62  e0 5b c3 15 01 00 00 00  0c 00 00 01 01 00 00 00  00 00 00 00 00 00 00 00  8a 07 00 00
              ^~~~~~~~~~~~~~~~~~~~~~~
              vptr for 'QPaintEvent'
```

**Pull request merged**